go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager.go

```diff
@@ -608,3 +608,25 @@ func (jsIPManager *JoinSwitchIPManager) ReleaseJoinLRPIPs(nodeName string) (err
 	}
 	return err
 }
+
+// NewL2SwitchManager initializes a new layer2 logical switch manager,
+// only manage subnet for the one specified switch
+func NewL2SwitchManager() *LogicalSwitchManager {
+	return &LogicalSwitchManager{
+		cache:    make(map[string]logicalSwitchInfo),
+		RWMutex:  sync.RWMutex{},
+		ipamFunc: NewL2IPAMAllocator,
```

Comment:
what if I don't want IPAM in my flat L2 ? I.e. I want to use something - other than OVN-K - to manage the IPs in my subnet (cloud-init on the VMs / network DHCP / etc).

@qinqon this use case interests you afaiu.

Reply:
At the end we may use use ovn-k IPAM capability but exposed as DHCP, in that case the only part not needed is the address at the pod's veth, but is true that a pure L2 architecture has to allow not setting addresses.

Reply:
discussed in the meeting.